### PR TITLE
How to run postinstall for custom npm packages

### PR DESCRIPTION
**Pull request by Artur-**
See https://github.com/vaadin/flow/pull/12512

### CODE REVIEW

File: articles/shared/guide/install/_pnpm.asciidoc [hunk 1]
```diff
@@ -36,3 +36,10 @@ See the https://pnpm.js.org/[pnpm website] for more information about available
 [NOTE]
 Vaadin expects transitive platform dependencies to be available directly under `node_modules`.
 In Vaadin versions earlier than 19 that use pnpm, the `--shamefully-hoist` flag must be explicitly given to pnpm during package installation: `pnpm i --shamefully-hoist mobx`.
+
+= Custom Packages that need postinstall scripts
+
+Starting from Vaadin 23, postinstall scripts for npm/pnpm packages are not run by default. This is to remove an attack vector where a new npm/pnpm package version with malware is published.
```
Comment (caalador) [inline articles/shared/guide/install/_pnpm.asciidoc:42]: So the feature shouldn't be backported to 14.x and 22?

File: articles/shared/guide/install/_pnpm.asciidoc [hunk 2]
```diff
@@ -36,3 +36,10 @@ See the https://pnpm.js.org/[pnpm website] for more information about available
 [NOTE]
 Vaadin expects transitive platform dependencies to be available directly under `node_modules`.
 In Vaadin versions earlier than 19 that use pnpm, the `--shamefully-hoist` flag must be explicitly given to pnpm during package installation: `pnpm i --shamefully-hoist mobx`.
+
+= Custom Packages that need postinstall scripts
```
Comment (jouni) [inline articles/shared/guide/install/_pnpm.asciidoc:40]:
```suggestion
= Custom Packages That need Post Install Scripts

```

File: articles/shared/guide/install/_pnpm.asciidoc (hunk 1 above)
Comment (Artur-) [reply articles/shared/guide/install/_pnpm.asciidoc:42]: I would consider V14 later. It probably needs more testing etc also as the dependencies are different. For instance `esbuild` which requires postinstall is not used there I think

File: articles/shared/guide/install/_pnpm.asciidoc [hunk 3]
```diff
@@ -36,3 +36,10 @@ See the https://pnpm.js.org/[pnpm website] for more information about available
 [NOTE]
 Vaadin expects transitive platform dependencies to be available directly under `node_modules`.
 In Vaadin versions earlier than 19 that use pnpm, the `--shamefully-hoist` flag must be explicitly given to pnpm during package installation: `pnpm i --shamefully-hoist mobx`.
+
+= Custom Packages that need postinstall scripts
+
+Starting from Vaadin 23, postinstall scripts for npm/pnpm packages are not run by default. This is to remove an attack vector where a new npm/pnpm package version with malware is published.
+
+Some legitimate packages still use postinstall scripts and rely on them to work properly. For internal dependencies, Vaadin runs the postinstall scripts automatically. If you add custom packages that need postinstall scripts to be run, you can add the packages as a comma separated list to the `vaadin.npm.postinstallPackages` system property (for development mode) and using the `<postinstallPackages>` configuration property in Maven/Gradle for production mode.
```
Comment (jouni) [inline articles/shared/guide/install/_pnpm.asciidoc:44]:
```suggestion
Some legitimate packages still use `postinstall` scripts and rely on them to work properly.
For internal dependencies, Vaadin runs the `postinstall` scripts automatically.

You can allow custom packages to run `postinstall` scripts if needed.
For development mode, add the packages as a comma separated list to the `vaadin.npm.postinstallPackages` system property.
For production builds, add the `<postinstallPackages>` Maven/Gradle configuration property.
```

**Comment by Artur-** [comment]
The term used in npm docs is `postinstall` and not `post install`. Do you still think we should name them "Post Install" scripts in the header?

**Comment by jouni** [comment]
> The term used in npm docs is postinstall and not post install. Do you still think we should name them "Post Install" scripts in the header?

Headings should prefer not using "code" formatting/terms. Not sure if "Post-Install" is more correct language-wise.

With "Post Install", search results aren't very good if you search for "postinstall". 

I suppose we could add the term "postinstall" to accept.txt, and then disable the heading check for this particular heading:

```asciidoc

pass:[<!-- Vale Vaadin.Headings = NO -->]

== “postinstall” Scripts

pass:[<!-- Vale Vaadin.Headings = YES -→]
```

**Comment by Artur-** [comment]
I can live with "Post Install" as it says "postinstall" in the content

File: articles/shared/guide/install/_pnpm.asciidoc [hunk 4]
```diff
@@ -36,3 +36,18 @@ See the https://pnpm.js.org/[pnpm website] for more information about available
...
+
+[role="since:com.vaadin:vaadin@V23"]#No `postinstall` scripts for npm/pnpm packages are run by default#.
+There are known incidents of new package versions containing malware which have abused `postinstall` scripts.
+Not running them removes that security risk.
+
+Some legitimate packages still use `postinstall` scripts and rely on them to work properly.
+For internal dependencies, Vaadin runs the `postinstall` scripts automatically.
+
+You can allow custom packages to run `postinstall` scripts if needed.
+For development mode, add the packages as a comma separated list to the `vaadin.npm.postinstallPackages` system property.
```
Comment (dougjewsbury) [inline articles/shared/guide/install/_pnpm.asciidoc:51]:
```suggestion
For development mode, add the packages as a comma-separated list to the `vaadin.npm.postinstallPackages` system property.

```

**Review by jouni — APPROVED** [review]

**Comment by jouni** [comment]
The Netlify preview failed because of a misconfiguration I did.

**Pull request merged**